lib/debugger/modes/commandRepl.js

```diff
@@ -30,7 +30,13 @@ var CommandRepl = function(client) {
 CommandRepl.prototype.stepEval = function(expression, callback) {
   expression = expression.replace(/"/g, '\\\"');
   var expr = 'browser.dbgCodeExecutor_.execute("' + expression + '")';
-  this.evaluate_(expr, callback);
+  this.evaluate_(expr, function(err, res) {
+    // Result is a string representation of the evaluation.
+    if (res !== undefined) {
+      console.log(res);
```

Comment:
Before I was using JSON.stringify to serialize the data between protractor and the commandRepl, which would parse the string. However, this was not good because you can't serialize functions and objects with circular dependencies. 

The only way to stringify the full object mimic what `console.log` does, which is to use `util.format` (see https://github.com/joyent/node/blob/v0.12.0/lib/console.js#L55). 

So then, lastly, why I can't do `callback(err, res)`, but must instead do `console.log(res); callback(err, undefined)`. It's because during callback, nodejs doesn't console.log the result, but will rather print the result as is. 

Here are some outputs if I do `callback(err, res)`:

```
> browser
'{ controlFlow: [Function],\n  schedule: [Function],\n  getSession: [Function],\n  getCapabilities: [Function],\n  quit: [Function],\n  actions: [Function],\n  executeScript: [Function],\n  executeAsyncScript: [Function],\n  call: [Function],\n  wait: [Function],\n  sleep: [Function],\n  getWindowHandle: [Function],\n  getAllWindowHandles: [Function],\n  getPageSource: [Function],\n  close: [Function],\n  getCurrentUrl: [Function],\n  getTitle: [Function],\n  findElementInternal_: [Function],\n  findDomElement_: [Function],\n  findElementsInternal_: [Function],\n  takeScreenshot: [Function],\n  manage: [Function],\n  switchTo: [Function],\n  driver: \n   { session_: \n      { then: [Function: then],\n        cancel: [Function: cancel],\n        isPending: [Function: isPending] },\n     executor_: { execute: [Function] },\n     flow_: \n      { events_: {},\n        timer: [Object],\n        history_: [],\n        activeFrame_: [Object],\n        schedulingFrame_: [Object],\n        shutdownId_: null,\n        eventLoopId_: [Object],\n        pendingRejections_: 0,\n        numAbortedFrames_: 0 } },\n  element: { [Function] all: [Function] },\n  \'$\': [Function],\n  \'$$\': [Function],\n  baseUrl: \'http://www.angularjs.org\',\n  rootEl: \'body\',\n  ignoreSynchronization: false,\n  getPageTimeout: 10000,\n  params: {},\n  resetUrl: \'data:text/html,<html></html>\',\n  mockModules_: [ { name: \'protractorBaseModule_\', script: [Function], args: [] } ],\n  getProcessedConfig: [Function],\n  forkNewDriverInstance: [Function],\n  dbgCodeExecutor_: \n   { execPromise_: \n      { then: [Function: then],\n        cancel: [Function: cancel],\n        isPending: [Function: isPending] },\n     execPromiseResult_: undefined,\n     execPromiseError_: undefined,\n     replServer_: \n      { domain: null,\n        _events: {},\n        _maxListeners: 10,\n        useGlobal: false,\n        ignoreUndefined: false,\n        eval: [Function],\n        inputStream: [Object],\n        outputStream: [Object],\n        lines: [Object],\n        context: [Object],\n        bufferedCommand: \'\',\n        prompt: \'> \',\n        rli: [Object],\n        commands: [Object],\n        writer: [Object],\n        useColors: false },\n     execute_: [Function],\n     execute: [Function],\n     complete: [Function],\n     resultReady: [Function],\n     getResult: [Function] } }'
> element
'function (locator) {\n    return new ElementArrayFinder(ptor).all(locator).toElementFinder_();\n  }'
```